model/map/src/main/java/org/keycloak/models/map/user/MapUserAdapter.java

```diff
@@ -211,7 +211,14 @@ public String getEmail() {
     @Override
     public void setEmail(String email) {
         email = KeycloakModelUtils.toLowerCaseSafe(email);
-        if (email != null && email.equals(entity.getEmail())) return;
+        if (email != null) {
+            if (email.equals(entity.getEmail())) {
+                return;
+            }
+            if (email.trim().isEmpty()) {
```

Comment:
Aren't there any StringUtils which allow to write these three lines as simple oneliner like email = StringUtils.trimmToNull(email); ?

Reply:
There isn't. But they are coming in a different PR. We can refactor that later.